woocommerce/wc-api-dev
# Fix plugin auto update logic

Auto updates were previously added in #16.

There was a problem with the auto-update logic. It worked with the testing instructions I provided (i.e. manually running `wp_maybe_auto_update()`), but doesn't seem to work out in the wild :(.

I've spent quite a bit of time trying to track it down and I think I've discovered why.

The relevant part seems to be where we bail early for on `$transient->checked` around `line 60`.  This doesn't seem to always be set - specifically in the case of [`DOING_CRON`](https://github.com/WordPress/WordPress/blob/064e62cbeac39a062a46a7d2d7e94b297ad5e8c0/wp-includes/update.php#L237).  

I also looked at what our very own Woo Helper is doing, as well as a couple other GH update plugins and they don't bail like this. They check against the internal version of the plugin and add to the list of plugins to be updated.

The other addition here I made is storing the raw GitHub response (containing the latest release) in a transient. It's a short lived transient, but I wanted to avoid making multiple calls to the endpoint in rapid succession.

To Test (preferably on your AT site):
* Download, install, and activate this helper plugin: 
[reschedule-version-check.zip](https://github.com/woocommerce/wc-api-dev/files/1191818/reschedule-version-check.zip). It allows you to easily schedule the next version check to make testing easier, and will send a WordPress debug email with auto-update results.
* Download 
https://cloudup.com/c5PYieevCBA
 which contains a zipped copy of this branch. There is one different between this zip and the PR here. I have set the version to `0.7.9`, so that when you are testing, the version will be seen as out of date.
* Upload this plugin over your existing `wc-api-dev` plugin and activate.
* Go to `https://:site?version-check=reschedule` which should schedule a version check/update for 2 minutes from now.
* Wait for two minutes.
* Visit any page on your site, like the front page. WP's cron system will kick in.

You should receive an email like this: 

![notice](https://user-images.githubusercontent.com/689165/28845252-3ee57478-76bc-11e7-9ba5-2566bd5a849f.png)
